### DISCUSSION TITLE
`yupResolver` package cannot work with async test cases

### DISCUSSION THREAD

**Discussion by sitole** [mantinedev/mantine#3245] [Feature requests]
### What package has an issue

@mantine/form

### Describe the bug

Hi!
i'm trying to implement async Yup validation for my form.
Basically have validation function that must be promise (heavy phone library loaded)
with Yup its okay to use async function in validation like this:

```javascript
const schema = yup.object().shape({
    yup
        .string()
        .test('phone', 'Phone is not valid format', isPhoneValid)
});
```
and `isPhoneValid` is promise function that returns boolean.
With Yup it works well but when I want to integrate Mantine format and Yup with `validate: yupResolver(schema)`
and try to validate my form there is React error message:

<img width="977" alt="image" src="https://user-images.githubusercontent.com/31341350/209569068-1352f2d7-6899-4324-ac35-7cc904f2ac9a.png">

Looks like it's inside of `yupResolved` method. This method also looks like it's written only for sync validation. (https://github.com/mantinedev/mantine/blob/c7d080c2133b0196e3a8382ec6134838632c8f9a/src/mantine-form/src/resolvers/yup-resolver/yup-resolver.ts#L21)

its bug? If not how I should use async validation with Mantine yup adapter?
Thanks 🚀

### What version of @mantine/hooks page do you have in package.json?

^5.9.5

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
It is expected behavior – currently `@mantine/form` package does not support async validation.

**Reply by sitole** [reply]
@rtivital is there any chance how I can use a sync validation? For example verify if blog post slug is free etc?

I mean I can build it custom in form but some support directly for form would be amazing 🧡

**Reply by rtivital** [reply]
No, async validation is currently not supported, I am not aware of any methods that you can use for this
